editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -1522,18 +1522,30 @@ int TileMapLayerEditorTilesPlugin::_get_transformed_alternative(int p_alternativ
 	bool transform_flip_v = p_alternative_id & TileSetAtlasSource::TRANSFORM_FLIP_V;
 	bool transform_transpose = p_alternative_id & TileSetAtlasSource::TRANSFORM_TRANSPOSE;
 
+	bool is_hexagon = _get_edited_layer()->get_tile_set()->get_tile_shape() == TileSet::TILE_SHAPE_HEXAGON;
+
 	switch (p_transform) {
 		case TRANSFORM_ROTATE_LEFT: { // (h, v, t) -> (v, !h, !t)
```

Comment:
Note that the changes make this comment partially outdated (same for `case TRANSFORM_ROTATE_RIGHT`), as it refers only to 90° rotations, which now is not what would be always performed.